ext/MooncakeNNlibExt.jl

```diff
@@ -0,0 +1,66 @@
+module MooncakeNNlibExt
+
+using NNlib, Random, Mooncake
+using Base: IEEEFloat
+using NNlib: dropout
+
+using NNlib: conv, depthwiseconv
+import Mooncake: @from_rrule, DefaultCtx, MinimalCtx
+
+@from_rrule(
+    MinimalCtx,
+    Tuple{typeof(batched_mul), Array{P, 3}, Array{P, 3}} where {P<:IEEEFloat},
+)
+@from_rrule(
+    MinimalCtx,
+    Tuple{typeof(dropout), AbstractRNG, Array{P}, P} where {P<:IEEEFloat},
+    true,
+)
+@from_rrule(MinimalCtx, Tuple{typeof(softmax), Array{<:IEEEFloat}}, true)
+@from_rrule(MinimalCtx, Tuple{typeof(logsoftmax), Array{<:IEEEFloat}}, true)
+@from_rrule(MinimalCtx, Tuple{typeof(logsumexp), Array{<:IEEEFloat}}, true)
+@from_rrule(
+    MinimalCtx,
+    Tuple{typeof(upsample_nearest), Array{<:IEEEFloat}, NTuple{N, Int} where {N}},
+)
+@from_rrule(
+    MinimalCtx,
+    Tuple{
+        typeof(NNlib.fold), Array{<:IEEEFloat}, NTuple{N, Int} where {N}, DenseConvDims,
+    },
+)
+@from_rrule(
+    MinimalCtx, Tuple{typeof(NNlib.unfold), Array{<:IEEEFloat}, DenseConvDims}
+)
+@from_rrule(
+    MinimalCtx,
+    Tuple{typeof(NNlib.scatter), Any, Array, Array{<:Union{Integer, Tuple}}},
+    true,
+)
+for conv in [:conv, :depthwiseconv]
+    local ∇conv_data, ∇conv_filter = Symbol.(:∇, conv, [:_data, :_filter])
+
+    @eval @from_rrule(
+        MinimalCtx,
+        Tuple{typeof($conv), Array{P}, Array{P}, ConvDims} where {P<:IEEEFloat},
+        true,
+    )
+    @eval @from_rrule(
+        MinimalCtx,
+        Tuple{typeof($∇conv_data), Array{P}, Array{P}, ConvDims} where {P<:IEEEFloat},
+        true,
+    )
+end
+@eval @from_rrule(
```

Comment:
This was a mistake on my part. `@eval` is only needed when I'm iterating over symbols and interpolating them. So the for loops above and below this line need `@eval` for interpolation purposes. Here, I'm not interpolating anything, so it can go.